projects/views.py

```diff
@@ -148,26 +108,31 @@ def get(self, request, project_id, *args, **kwargs):
         serializer = serializer_class(page_obj, many=True)
         project_serializer = ProjectListSerializer(project_instance, many=False)
 
-        type_list = (
-            project_instance.components.all().order_by().values_list("type").distinct()
-        )
+        type_list = project_instance.components.all().order_by().values_list("type").distinct()
 
-        response = []
-        response.append({"project": project_serializer.data})
-        response.append({"components": serializer.data})
-        response.append({"total_item_count": paginator.count})
-        response.append({"type_list": type_list})
+        response = [
+            {"project": project_serializer.data},
+            {"components": serializer.data},
+            {"total_item_count": paginator.count},
+            {"type_list": type_list}
+        ]
 
         return Response(response, status=status.HTTP_200_OK)
 
 
-class ProjectComponentNotAddedListView(APIView):
-    def get(self, request, project_id, *args, **kwargs):
-        project_instance = get_object_or_404(Project, pk=project_id)
+class ProjectComponentNotAddedListView(generics.GenericAPIView):
+    queryset = Project.objects.all()
+    lookup_url_kwarg = "project_id"
+    serializer_class = ComponentListBasicSerializer
+
+    def get(self, request, *args, **kwargs):
```

Comment:
Need to overwrite `list` instead of `get`. I'll update.

Reply:
Nevermind. I think this would require a more significant refactor, so I'll hold on this one.

Reply:
I lied, I fixed it.

Reply:
Nope, still needs more of a refactor. Excluding this for this PR.